### PR TITLE
fail iothread tests when there is not enough cpu resources

### PR DESCRIPTION
**Pull request by mfranczy**
Signed-off-by: Marcin Franczyk <mfranczy@redhat.com>

**What this PR does / why we need it**:
It fails iothread tests in case there is not enough cpu resources

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: tests/vmi_iothreads_test.go [hunk 1]
```diff
@@ -133,9 +133,7 @@ var _ = Describe("IOThreads", func() {
 		})
 
 		table.DescribeTable("[ref_id:2065] should honor auto ioThreadPolicy", func(numCpus int, expectedIOThreads int) {
-			if numCpus > availableCPUs {
-				Skip(fmt.Sprintf("Testing environment does not contain a node with required %d CPUs number, the highest detected number is %d", numCpus, availableCPUs))
-			}
+			Expect(numCpus <= availableCPUs).To(BeTrue(), "Testing environment should have at least one node with required CPUs number")
```
Comment (dhiller) [inline tests/vmi_iothreads_test.go]: I would recommend to put the number of required CPUs into the message, i.e. reuse the `fmt.Sprintf` that you used before. This way the recipient exactly knows what the requirements are. I.e. sth like:

    Expect(numCpus <= availableCPUs).To(BeTrue(), 
        fmt.Sprintf("Testing environment only has nodes with %d CPUs available, but required are %d CPUs", availableCPUs, numCpus))

Comment (rmohr) [reply tests/vmi_iothreads_test.go]: :+1:

Comment (mfranczy) [reply tests/vmi_iothreads_test.go]: good point, done.

**Comment by mfranczy** [comment]
sorry I forgot to add `,` at the end of the line here https://github.com/kubevirt/kubevirt/pull/2734/files#diff-9dfd115e7d7e9c5f1eda678d02ccab9aR138, it's fixed now.

**Comment by mfranczy** [comment]
and I just realized that I missed multi queue tests, I think it's better to add it here than open a new PR. Anyway, I won't add anything new here.

File: tests/vmi_iothreads_test.go [hunk 2]
```diff
@@ -133,9 +133,9 @@ var _ = Describe("IOThreads", func() {
 		})
 
 		table.DescribeTable("[ref_id:2065] should honor auto ioThreadPolicy", func(numCpus int, expectedIOThreads int) {
-			if numCpus > availableCPUs {
-				Skip(fmt.Sprintf("Testing environment does not contain a node with required %d CPUs number, the highest detected number is %d", numCpus, availableCPUs))
-			}
+			Expect(numCpus <= availableCPUs).To(BeTrue(),
```
Comment (slintes) [inline tests/vmi_iothreads_test.go]: nit: `BeNumerically()` is a bit nicer
https://onsi.github.io/gomega/#benumericallycomparator-string-compareto-interface

File: tests/vmi_multiqueue_test.go [hunk 3]
```diff
@@ -52,8 +52,13 @@ var _ = Describe("MultiQueue", func() {
 
 	Context("MultiQueue Behavior", func() {
 
+		availableCPUs := tests.GetHighestCPUNumberAmongNodes(virtClient)
+
 		It("[test_id:959][rfe_id:2065] Should honor multiQueue requests", func() {
 			numCpus := 3
+			Expect(numCpus <= availableCPUs).To(BeTrue(),
```
Comment (slintes) [inline tests/vmi_multiqueue_test.go]: same

File: tests/vmi_iothreads_test.go (hunk 2 above)
Comment (mfranczy) [reply tests/vmi_iothreads_test.go]: sure, done.

File: tests/vmi_multiqueue_test.go (hunk 3 above)
Comment (mfranczy) [reply tests/vmi_multiqueue_test.go]: done

**Pull request merged**